### ISSUE TITLE
`BatchNormalization` does not transform the input at all.

### ISSUE THREAD

**Issue by khteh** [tensorflow/tensorflow#102870]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.20.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 25.04

### Python version

3.13.3

### Current behavior?

What’s the equivalent of PyTorch `InstanceNorm2d(channels)`? I use `BatchNormalization(axis=1)` and using it with input of shape `(n_test, image_channels, image_size, image_size)` doesn’t seem to transform anything at all.

### Standalone code to reproduce the issue

```shell
instance_norm = BatchNormalization(axis=1)
test_input = numpy.ones((n_test, image_channels, image_size, image_size))
test_input[:, :, 0] = 0
normalized_image = instance_norm(image)
print(f"image: {tf.math.reduce_mean(image)}, normalized_image: {tf.math.reduce_mean(normalized_image)}")
```

### Relevant log output

```shell
image: 0.6666666666666666, normalized_image: 0.6663335561752319
```

**Comment by Venkat6871** [comment]
Hi @khteh,
Apologies for the delay, and thank you for raising this issue.
The behavior you observed is expected `tf.keras.layers.BatchNormalization` and `PyTorch’s` InstanceNorm2d are conceptually different.
According to the [official Keras documentation](https://keras.io/api/layers/normalization_layers/batch_normalization/), `BatchNormalization` normalizes activations across the batch (subtracting the batch mean and dividing by the batch standard deviation). It aggregates statistics across the batch and does not normalize each sample individually.
In contrast, PyTorch’s `InstanceNorm2d(channels)` normalizes each instance independently across spatial dimensions for each channel. That’s why in your example:
```
instance_norm = BatchNormalization(axis=1)
```
the output mean remains nearly the same BatchNorm aggregates batch-wide statistics, so similar samples share nearly identical normalization.

To replicate `InstanceNorm2d` behavior in TensorFlow, you can use tf.keras.layers.GroupNormalization with groups=channels, which normalizes each channel independently. Here is the [gist](http://colab.sandbox.google.com/gist/Venkat6871/3840ac56d0a981c47646ffdd477df984/102870_tf_2-19-0-2-20-0-v.ipynb) for your reference. This achieves the same normalization behavior as PyTorch’s InstanceNorm2d(channels). You can also refer to the [GroupNormalization documentation](https://keras.io/api/layers/normalization_layers/group_normalization/) for more details.

Thank you!
